### ISSUE TITLE
[Bug]: milvus 2.0.2  the relation between pk and vector is not consistent after excute cmd list_collection

### ISSUE THREAD

**Issue by wj-0116** [milvus-io/milvus#17505]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: 2.0.2
- Deployment mode(standalone or cluster): cluster
- SDK version(e.g. pymilvus v2.0.0rc2): go sdk
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

at the beginning, insert（pk、vector），the result is：
![image](https://user-images.githubusercontent.com/107320535/173212766-0a56707a-ce0f-4d87-9833-0f8e47537406.png)
after execute list_collection，the result changed：
![image](https://user-images.githubusercontent.com/107320535/173212794-12ef6aa9-081b-4334-a5ed-738478d390b5.png)

### Expected Behavior

the relation between pk and vector should keep consistent

**Comment by xiaofan-luan** [comment]
@wj-0116 Thanks for your issue, please change to english for the convenience of non-chinese speakers, thanks !

**Comment by wj-0116** [comment]
> @wj-0116 Thanks for your issue, please change to english for the convenience of non-chinese speakers, thanks !

have changed to english

**Comment by xiaofan-luan** [comment]
Thanks, we will investigate on it.

**Comment by congqixia** [comment]
Hi @wj-0116 Thanks for the issue! Could you please provide the source code to reproduce this problem(with data omit if sensitive)?

**Comment by wj-0116** [comment]
client code:
package dao

import (
	"context"
	"git.garena.com/shopee/platform/splog"
	"github.com/milvus-io/milvus-sdk-go/v2/client"
	"github.com/milvus-io/milvus-sdk-go/v2/entity"
	"log"
	"originalsearchcontroller/config"
	"originalsearchcontroller/internal/db"
)

var (
	vectorField     string
	primaryKeyField string
)

func GetCollectionInfo() (pk, vectorField string) {
	conf := config.GetConfig()
	result, err := db.MilvusClient.DescribeCollection(context.Background(), conf.MilvusConfig.Collection)
	if err != nil {
		log.Panicln(err)
	}
	for _, v := range result.Schema.Fields {
		if v.DataType == entity.FieldTypeFloatVector || v.DataType == entity.FieldTypeBinaryVector {
			//e = v.DataType
			vectorField = v.Name
			indexInfo, err := db.MilvusClient.DescribeIndex(context.Background(), conf.MilvusConfig.Collection, v.Name)
			if err != nil {
				log.Panicln(err)
			}
			if len(indexInfo) < 1 {
				log.Fatalf("coll: %s not index\n", conf.MilvusConfig.Collection)
			}
		}
		if v.PrimaryKey == true {
			pk = v.Name
		}
	}
	splog.WithFields(splog.Fields{
		"pk":          pk,
		"VectorField": vectorField,
	}).Info("GetCollectionInfo: ")
	return
}

func Search(ctx context.Context, collection string, topN int, nprobe int, vectors []float32) ([]client.SearchResult, error) {
	if primaryKeyField == "" {
		primaryKeyField, vectorField = GetCollectionInfo()
	}
	sp, _ := entity.NewIndexFlatSearchParam(
		nprobe, // searchParam
	)
	return db.MilvusClient.Search(ctx, collection, []string{}, "",
		[]string{primaryKeyField}, []entity.Vector{entity.FloatVector(vectors)}, vectorField, entity.IP, topN, sp)
}

func InsertOne(ctx context.Context, collection string, vectors []float32, ids int64) error {
	idColumn := entity.NewColumnInt64(primaryKeyField, []int64{ids})
	vectorColumn := entity.NewColumnFloatVector(vectorField, len(vectors), [][]float32{vectors})
	_, err := db.MilvusClient.Insert(
		ctx,        // ctx
		collection, // CollectionName
		"",         // partitionName
		idColumn,   // columnarData
		vectorColumn,
	)
	return err
}

type MilvusItem struct {
	ID     int64
	Vector []float32
}

func BulkInsert(ctx context.Context, collection string, items []MilvusItem) error {
	var ids = make([]int64, len(items))
	var vecs = make([][]float32, len(items))
	for i := range items {
		ids[i] = items[i].ID
		vecs[i] = items[i].Vector
	}
	idColumn := entity.NewColumnInt64(primaryKeyField, ids)
	vectorColumn := entity.NewColumnFloatVector(vectorField, 512, vecs)
	_, err := db.MilvusClient.Insert(
		ctx,        // ctx
		collection, // CollectionName
		"",         // partitionName
		idColumn,   // columnarData
		vectorColumn,
	)
	return err
}

func ExistsMilvusByID(ctx context.Context, collection string, id int64) (bool, error) {
	values, err := db.MilvusClient.QueryByPks(ctx, collection, []string{}, entity.NewColumnInt64(primaryKeyField, []int64{id}), []string{primaryKeyField})
	return len(values) != 0, err
}

**Comment by wj-0116** [comment]
this issue can be reproduce using the following steps:
1. prepare id and vector mapping data in file.(eg. 100w)
2. split the data file into 20 subfile.
3. create collection
4. create index for collection
5. use 20 process to insert data to milvus. (each process read one subfile, and insert to the collection)
6. execute list_collection command during step.5 

sdk: pymilvus 2.0
operate: collection = Collection(collection_name, schema, consistency_level="Strong")、ret = collection.create_index("vector", index)、collection.insert(entities)、utility.list_collections()、result = collection.query(expr=expr, output_fields=["id", "vector"])

**Comment by wj-0116** [comment]
上传下，复现用的脚本，主要是pymilvus的操作

**Comment by congqixia** [comment]
@wj-0116 quick question, it seems that your pk value is user-defined value. Does it contains duplicated pk?

**Comment by wj-0116** [comment]
> @wj-0116 quick question, it seems that your pk value is user-defined value. Does it contains duplicated pk?

there is no duplicated pk, it is a auto-increament id generate by mysql.

**Comment by wj-0116** [comment]
data sample we used to reproduce the issue.  due to size limit, I just update a few sample data.

[data.10sample.txt](https://github.com/milvus-io/milvus/files/8923497/data.10sample.txt)

**Comment by yanliang567** [comment]
this should be fixed in 2.1. close it for now.
